fudge-ratio/java/servlets/FudgeRatio.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+import com.google.gson.Gson;
+import java.util.*; 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    }
+
+
+
+    private double calculateRatio(String start, String end, int expected){
+        //parse the starting time time string (in format 00:00:00 (hrs:min:second))
+        int startHr = Integer.parseInt(start.substring(0, 2));
```

Comment:
OPINION: Does startHr change? or is in any way expected to change?
Maybe just put a final into all the variables we do not expect to change to express that intent